File: LeetCodeRecord/MethodFirst/717-IsOneBitCharacter.java
package MethodFirst;

class IsOneBitCharacter {
    public boolean isOneBitCharacter(int[] bits) {
        int i = 0;
        while (i < bits.length - 1) {
            i += bits[i] + 1;
        }

        return i == bits.length - 1;
    }    
}

File: LeetCodeRecord/ACFirst-Golang/717-IsOneBitCharacter.go
package main

func isOneBitCharacter(bits []int) bool {
	i := 0

	for i < len(bits)-1 {
		if bits[i] == 1 {
			i += 2
		} else {
			i++
		}
	}

	return i == len(bits)-1
}
